### ISSUE TITLE
Way to access original labels?

### ISSUE THREAD

**Issue by nishamuktewar** [tristandeleu/pytorch-meta#85]
First of all, thank you for this library and even the MAML one. I have been experimenting with it since last couple of months and have been able to edit it and make it work on a new dataset. Thanks again!

Was wondering if it is currently possible or if there is some way to extend the existing code to store original labels in the dataloader process. What I mean by that is:

 Currently, the original data labels are available in the following way:
```
train_dataset = newdata(folder='./ndata',
                    shots=1,
                    ways=5,
                    shuffle=True,
                    test_shots=1,
                    meta_train=True,
                    download=False,
                    transform=train_transform)
```
Where `newdata` is a `CombinationMetaDataset`. And then access the `train_dataset` labels with `train_dataset.dataset.labels`. 
However, with the data loader 

```
train_dataloader = BatchMetaDataLoader(train_dataset,
                                 batch_size=1,
                                 shuffle=True,
                                 num_workers=1)
num_batch = 0
for batch in train_dataloader:
    if num_batch == 0:
        train_inputs, train_targets = batch["train"]
        print('Train inputs shape: {0}'.format(train_inputs.shape))    # (1, 5, 1, 28, 28)
        print('Train targets shape: {0}'.format(train_targets.shape))  # (1, 5)
        print(train_targets)
    else:
        break
```
and it's corresponding output `[1, 4, 0, 2, 3]` as shown below the labels reflect the `n-way` numbers (and rightly so) and not the original labels:

```
Train inputs shape: torch.Size([1, 5, 1, 28, 28])
Train targets shape: torch.Size([1, 5])
tensor([[1, 4, 0, 2, 3]])
```

And yes, this is how it should be with the meta-learning process but is there a way to still access the corresponding original labels? If not, is it possible to extend the code? Any pointers in this direction will be greatly appreciated.

**Comment by tristandeleu** [comment]
Thank you for the kind words! You can access the original labels using a custom `target_transform`, based on the default `Categorical` target transform which converts the original labels (+ their class augmentations if any) into n-way integers. Here is a small example:

```python
from torchmeta.datasets.helpers import omniglot
from torchmeta.utils.data import BatchMetaDataLoader
from torchmeta.transforms import Categorical

class CategoricalAndLabels(Categorical):
    def __call__(self, target):
        label, class_augmentation = target
        return (self.classes[target], label)

dataset = omniglot('data',
                   ways=5,
                   shots=5,
                   meta_train=True,
                   target_transform=CategoricalAndLabels(num_classes=5))

dataloader = BatchMetaDataLoader(dataset, batch_size=4)

batch = next(iter(dataloader))
train_inputs, train_targets = batch['train']

# Unpack the targets in [0, 4] and the original labels
train_targets, original_labels = train_targets

# Transpose the original labels
original_labels = list(zip(*original_labels))

print(f'train_target:\n{train_targets[0]}')
print(f'original_labels:\n{original_labels[0]}')

# train_target:
# tensor([2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 4, 4, 4, 4, 4, 3, 3, 3, 3,
#         3])
# original_labels:
# ('images_background/Arcadian/character04', 'images_background/Arcadian/character04',
#  'images_background/Arcadian/character04', 'images_background/Arcadian/character04',
#  'images_background/Arcadian/character04', 'images_background/Gujarati/character39',
#  'images_background/Gujarati/character39', 'images_background/Gujarati/character39',
#  ...
#  'images_background/Korean/character37', 'images_background/Korean/character37')
```

One small caveat: if you also want to return the class augmentations as well (if you happen to use class augmentations), you should probably return it as a string (simply using `str(class_augmentation)`) to work with PyTorch's collate function.
